src/Resources/config/services.yml

```diff
@@ -362,12 +359,22 @@ services:
         arguments:
             - "@security.http_utils"
             - "@contao.framework"
-            - "@router"
+            - "@translator"
+            - "@logger"
+
+    contao.security.frontend_authentication_success_handler:
+        class: Contao\CoreBundle\Security\Authentication\FrontendAuthenticationSuccessHandler
+        arguments:
+            - "@security.http_utils"
+            - "@contao.framework"
+            - "@translator"
+            - "@logger"
 
     contao.security.backend_user_provider:
-        class: Contao\CoreBundle\Security\User\BackendUserProvider
+        class: Contao\CoreBundle\Security\User\ContaoUserProvider
         arguments:
             - "@contao.framework"
+            - "\\Contao\\BackendUser"
```

Comment:
Can we at least pass `backend` or `frontend` and throw an exception if it is neither?

Reply:
What would that bring? We would need to `if-else` on the values to get the class?

Reply:
It would make it impossible to use the class with anything else than `backend` and `frontend`. You recall the initial discussion?

Reply:
Yeah, so in that case I can also check the two classes instead of introducing arbitrary names?